target/linux/mediatek/dts/mt7981b-asus-rt-ax57m.dts

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+#include "mt7981b.dtsi"
+/ {
+	model = "ASUS RT-AX57M";
+	compatible = "asus,rt-ax57m", "mediatek,mt7981";
+
+	aliases {
+		serial0 = &uart0;
+		label-mac-device = &gmac0;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-override = "";
+	};
+
+	memory@40000000 {
+		reg = <0 0x40000000 0 0x10000000>;
+		device_type = "memory";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&pio 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&pio 0 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&pio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		wan-blue {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&pio 5 GPIO_ACTIVE_LOW>;
+		};
+
+		wan-red {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&pio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			function = LED_FUNCTION_LAN;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&pio 7 GPIO_ACTIVE_LOW>;
+		};		
+
+		wlan24 {
+			function = LED_FUNCTION_WLAN_2GHZ;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&pio 8 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wlan5 {
+			function = LED_FUNCTION_WLAN_5GHZ;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&pio 9 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+
+	timer {
+		clock-frequency = <13000000>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&eth {
+	status = "okay";
+
+	gmac0: mac@0 {
+		/* LAN */
+		compatible = "mediatek,eth-mac";
+		reg = <0>;
+		nvmem-cells = <&macaddr_factory_4>;
+		nvmem-cell-names = "mac-address";
+		phy-mode = "2500base-x";
+
+		fixed-link {
+			speed = <2500>;
+			full-duplex;
+			pause;
+		};
+	};
+
+	gmac1: mac@1 {
+		/* WAN */
+		compatible = "mediatek,eth-mac";
+		reg = <1>;
+		nvmem-cells = <&macaddr_factory_4>;
+		nvmem-cell-names = "mac-address";
+		label = "wan";
+		phy-mode = "2500base-x";
+
+		fixed-link {
```

Comment:
This is wrong. The device is using the internal built-in 1GE PHY of the MT7981 SoC as WAN port. You should reference the internal PHY and also assign the correct MAC address.

Reply:
This is how it's defined in the official device tree and the Mac of the WAN is the same as LAN in stock firmware as well.

Regardless though, this was originally set as gigabit and that also doesn't work. See the change here for how it was originally https://github.com/openwrt/openwrt/pull/20491/commits/cc82826aa80d228478c2828a0e9219aa6089320f

Reply:
My best guess is on stock firmware they use phy-mode 2500base-x (like I know from my stock devicetree dump https://github.com/openwrt/openwrt/pull/20491#issuecomment-3465904148 , MACs redacted but they are the same) and use rate matching to support the lower link speeds. Without the firmware source yet I can't really say though.

Reply:
Well, you posted pictures of the board and there isn't any additional PHY chip apart from the MT7531 switch IC connected to GMAC0. So GMAC1 must be the internal PHY of the MT7981, so please try
```
	gmac1: mac@1 {
		compatible = "mediatek,eth-mac";
		reg = <1>;
		phy-mode = "gmii";
		phy-handle = <&int_gbe_phy>;
		nvmem-cells = <&macaddr_factory_4>;
		nvmem-cell-names = "mac-address";
		openwrt,netdev-name = "wan";
	};
```

Reply:
@dangowrt any thoughts on that ^ ? Let me know if I can help with anything while we are still waiting for stock source code (the Asus engineers are being slow to prepare it for customer support).

Reply:
I'm convinced that it must be the internal PHY, simply because there isn't any other PHY chip on that board. Try with my suggestion, but use `phy-mode = "internal";` and show me the complete bootlog with that.